lisa/util/shell.py

```diff
@@ -161,42 +160,37 @@ def try_connect(connection_info: ConnectionInfo, ssh_timeout: int = 300) -> Any:
                 key_filename=connection_info.private_key_file,
                 banner_timeout=10,
             )
+
+            stdin, stdout, _ = paramiko_client.exec_command("cmd\n")
```

Comment:
While line do you see the exception actually being? The paramiko_client is initialized in the loop. I think the key thing is to capture the `ConnectionResetError` exception. Is it right?

Reply:
This method is to use low level API (paramiko) to check the os type, and then use SpurShell to connect with the right shell type (windows or linux).

Reply:
This `paramiko_client.exec_command("cmd\n")` causes exceptions. I have seen it throwing both `SSHException` and `ConnectionResetError` error

Reply:
ok, we can go with this change, please rebase to latest main. Next time, you can create a new branch from main, and cherry pick it from original branch.

Reply:
Sure! Let me verify it for few other nested testcases before merging this